otcextensions/osclient/nat/v2/gateway.py

```diff
@@ -0,0 +1,252 @@
+            required=True,
+            help=_("Specifies the VPC ID"))
+        parser.add_argument(
+            '--internal-network-id',
+            metavar="<internal_network_id>",
+            required=True,
+            help=_("Specifies the network ID of thedownstream interface "
+                   "(the next hop ofthe DVR) of the NAT gateway."))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.nat
+
+        args_list = [
+            'name',
+            'description',
+            'spec',
+            'router_id',
+            'internal_network_id']
+        attrs = {}
+        for arg in args_list:
+            if getattr(parsed_args, arg):
+                attrs[arg] = getattr(parsed_args, arg)
+
+        obj = client.create_gateway(**attrs)
+
+        display_columns, columns = _get_columns(obj)
+        data = utils.get_item_properties(obj, columns)
+
+        return (display_columns, data)
+
+
+class UpdateNatGateway(command.ShowOne):
+    _description = _("Update a NAT Gateway.")
+
+    def get_parser(self, prog_name):
+        parser = super(UpdateNatGateway, self).get_parser(prog_name)
+        parser.add_argument(
+            'gateway',
+            metavar='<gateway>',
+            help=_('Specifies the Name or ID of the NAT Gateway.'),
+        )
+        parser.add_argument(
+            '--name',
+            metavar="<name>",
+            help=_("Specifies the name of the NAT Gateway."))
+        parser.add_argument(
+            '--description',
+            metavar="<description>",
+            help=_("Provides supplementary informationabout the NAT gateway."))
+        parser.add_argument(
+            '--spec',
+            metavar="<spec>",
+            help=_("Specifies the type of the NAT gateway."))
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.nat
+        args_list = [
+            'name', 'description', 'spec'
+        ]
+        attrs = {}
+        for arg in args_list:
+            if getattr(parsed_args, arg):
+                attrs[arg] = getattr(parsed_args, arg)
+        nat_gateway = client.find_gateway(parsed_args.gateway)
+
+        obj = client.update_gateway(nat_gateway.id, **attrs)
+
+        display_columns, columns = _get_columns(obj)
+        data = utils.get_item_properties(obj, columns)
+
+        return (display_columns, data)
+
+
+class DeleteNatGateway(command.Command):
+
+    _description = _("Deletes NAT Gateway.")
+
+    def get_parser(self, prog_name):
+        parser = super(DeleteNatGateway, self).get_parser(prog_name)
+        parser.add_argument(
+            'gateway',
+            metavar='<gateway>',
+            help=_('Specifies the Name or ID of the NAT gateway.'),
+        )
+
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.nat
+        nat_gateway = client.find_gateway(parsed_args.gateway)
```

Comment:
What do you expect if GW is not found? You try to delete not existing resource, since find defaults to "ignore_missing=False"

Reply:
Fixed in find_gateway method, now it will raise RuntimeError
obj = self._find(_gateway.Gateway, name_or_id)
```
        if obj is None:
            raise RuntimeError(
                'The provide NAT Gateway id/name={} '
                'could not be found'.format(name_or_id))
        return obj
```